File: HELLO_DJ_EMP/HELLO_DJ_EMP/daoemp.py
import pymysql
class DaoEmp:
    
    def __init__(self): #생성자
        self.conn = pymysql.connect(host='localhost', user='root', password='python',
                       db='python', port=3305, charset='utf8')
 
        self.curs = self.conn.cursor(pymysql.cursors.DictCursor)
        
    def selectList(self):
        sql = "select * from emp"
        self.curs.execute(sql)
         
        list = self.curs.fetchall()
        return list  
    
    def selectOne(self, e_id):
        sql = f"""
            select * from emp 
            where 
                e_id='{e_id}'
        """
        self.curs.execute(sql)
        
        # vos = self.curs.fetchall()
        vo = self.curs.fetchone()
        return vo
    
    def insert(self,e_id,e_name,gen,addr):
        sql = f"""
            insert into emp
            (e_id, e_name, gen, addr)
            values
            ('{e_id}', '{e_name}', '{gen}', '{addr}')
        """
        
        cnt = self.curs.execute(sql)
        self.conn.commit()
        return cnt
    
    def update(self,e_id,e_name,gen,addr):
        sql = f"""
            update emp
            set
                e_name='{e_name}',
                gen='{gen}',
                addr='{addr}'
            where 
                e_id='{e_id}'    
        """        
        cnt = self.curs.execute(sql)
        self.conn.commit()
        return cnt
    
    def delete(self,e_id):
        sql = f"""
            delete from emp
            where e_id='{e_id}'   
        """      
        cnt = self.curs.execute(sql)
        self.conn.commit()
        return cnt    
    
    

    def __del__(self):
        self.curs.close()
        self.conn.close()   
        
        
if __name__ == '__main__':
        de = DaoEmp()
        cnt = de.delete('5')
        print(cnt)
    

File: HELLO_EXPRESS_EMP/daoemp.js
var Mysql = require('sync-mysql');

class DaoEmp {
	
  	constructor() {
		this.conn = new Mysql({
		  host     : 'localhost', 	//db접속 주소
		  user     : 'root', 		//db접속id
		  password : 'python', 		//db접속pw
		  port	   : '3305',		//port번호
		  database : 'python' 		//db명
		});	
  	}

	selectList(){
		var sql = "SELECT * FROM emp";
		return this.conn.query(sql);
	}
	
	selectOne(e_id){
		var sql = `SELECT * FROM emp WHERE e_id = ${e_id}`;
		return this.conn.query(sql)[0];
	}
	
	insert(e_id, e_name, gen, addr){
		var sql = `insert into emp
						(e_id, e_name, gen, addr) 
					values
						('${e_id}','${e_name}','${gen}','${addr}')`;
		return this.conn.query(sql).affectedRows;
	}
	
	update(e_id, e_name, gen, addr){
		var sql = `UPDATE emp 
					SET 
						e_name = '${e_name}',
						gen = '${gen}',
						addr = '${addr}'
					WHERE 
						e_id='${e_id}'`;
		return this.conn.query(sql).affectedRows;
	}
	
	delete(e_id){
		var sql = `DELETE FROM emp 
					WHERE e_id='${e_id}'`;
		return this.conn.query(sql).affectedRows;
	}
}

module.exports = DaoEmp;

if(require.main === module){
	var de = new DaoEmp();
	//var cnt = de.delete("7");
	var list = de.selectList();
	//console.log("cnt",cnt);
	console.log("list",list);
}